EIPS/eip-7087.md

```diff
@@ -0,0 +1,75 @@
+---
+eip: 7087
+title: MIME Type For Web3 URL in Auto Mode
+description: Deteremine the MIME type of a Web3 URL in auto mode
+author: Qi Zhou (@qizhou), Nand2 (@nand2)
+discussions-to: https://ethereum-magicians.org/t/eip-7087-mime-type-for-web3-url-in-auto-mode/14471
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-05-28
+requires: 4804
+---
+
+## Abstract
+
+This standard extends the Web3 URL like `web3://` in [ERC-4804](./eip-4804.md) in auto mode to deteremine the MIME type of a link via additional queries.
+
+## Motivation
+
+Web3 URL defines a way to deteremine the MIME type of a link in auto mode via the suffix of the last argument.  However, most of existing contracts on Ethereum do not expect such suffix and thus the current Web3 URL cannot display the web contents on existing contracts in a correct way (e.g., rendering an SVG image rather than the XML text).  The standard proposes additional queries for a Web3 URL to deteremine the MIME type.
```

Comment:
```suggestion
Web3 URL defines a way to determine the MIME type of a link in auto mode via the suffix of the last argument.  However, most of existing contracts on Ethereum do not expect such suffix and thus the current Web3 URL cannot display the web contents on existing contracts in a correct way (e.g., rendering an SVG image rather than the XML text).  The standard proposes additional query parameters for a Web3 URL to determine the MIME type.
```